iosApp/Survey/Sources/Presentation/Coordinator/AppCoordinator.swift

```diff
@@ -23,7 +23,7 @@ struct AppCoordinator: View {
             case .splash:
                 SplashView(coordinator: coordinator)
             case .surveySelection:
-                SurveySelectionContainerView()
+                SurveySelectionContainerView(coordinator: coordinator)
             }
         }
```

Comment:
I think we can have this
```suggestion
        }
        .environmentObject(coordinator)
```

Reply:
Coordinator is `AccountCoordinator`, a specific type of coordinator. This way we allow only the necessary methods to be called in each View. Current `environmentObject` does not support passing protocol.

Reply:
@blyscuit I have another concern that the RouteCoordinator() instance in this file is created as `@ObservedObject` but do we really need to reset it every time updating the view? Shouldn't it be a `@StateObject` instead? 🤔

Reply:
@minhnimble [This](https://github.com/johnpatrickmorgan/FlowStacks#using-view-models) seems to be the way that the document show. I don't see an issue with using `StateObject` here from my testing and will change to use it. 

Please note that this view should be the root view so it will not be redraw by the effect of another view, hence the `ObservedObject` will not be reset.